components/brave_vpn/brave_vpn_service.cc

```diff
@@ -119,14 +114,28 @@ void BraveVpnService::GetHostnamesForRegion(ResponseCallback callback,
   base::Value dict(base::Value::Type::DICTIONARY);
   dict.SetStringKey("region", region);
   std::string request_body = CreateJSONRequestBody(dict);
-  OAuthRequest(base_url, "POST", request_body, false,
-               std::move(internal_callback));
+  OAuthRequest(base_url, "POST", request_body, std::move(internal_callback));
+}
+
+void BraveVpnService::GetProfileCredentials(
+    ResponseCallback callback,
+    const std::string& subscriber_credential,
+    const std::string& hostname) {
+  auto internal_callback =
+      base::BindOnce(&BraveVpnService::OnGetResponse, base::Unretained(this),
```

Comment:
are we 100% sure, that `BraveVpnService` will be alive when `internal_callback`?
`base::Unretained` may be danger

Reply:
@AlexeyBarabash good catch. i worked on it in january and many things have changed since then. updating it in a commit